### ISSUE TITLE
tutorial_isaac_gym shows poor performance

### ISSUE THREAD

**Issue by feidieufo** [AI4Finance-Foundation/ElegantRL#133]
I used the default parameters in the file to run the Ant and Humanoid environment, but the performance is not good, the reward can only be around 1000
![plot_learning_curve](https://user-images.githubusercontent.com/34419187/161894536-0631d6e3-2a21-44f5-936d-4a350c670796.jpg)
But the official rl_games code performance is good
![image](https://user-images.githubusercontent.com/34419187/161894741-c601e24e-43fe-48e5-836e-38ce212e5dfb.png)

**Comment by hmomin** [comment]
Hi, I'm one of the people who helped migrate the codebase from Isaac Gym Preview 2 to the latest release. I noticed the same thing as I was testing Isaac Gym on my machine.

I'm not sure who coded the original implementation, but I can suggest a few tips to improve it:

1. Don't use lists or numpy arrays for anything. Everything should be done using PyTorch tensors with `cat`, `stack`, `reshape`, etc. operations done directly on the tensors. Lists and numpy arrays are too slow, since they operate on the CPU, even if their contents are on the GPU.
2. The implementation in this repo makes use of multiprocessing to run the training and evaluation environments simultaneously as two separate processes/windows (see the image below). This is extremely inefficient from a memory standpoint: an entire rendering process with ground planes and physics is created only to run a single Ant environment for evaluation. Instead of having two separate processes (for example, training with 1024 environments and evaluation with 1 environment), why not just make use of a single process with, say, 2048 environments and choose one of the environments for evaluation? It would probably use roughly the same amount of memory and give you training samples twice as fast...

![isaac_gym_multiprocess](https://user-images.githubusercontent.com/72412509/162290783-94838772-a5c8-4a7a-ba75-872ab7518ad2.png)

I would try to help out with these improvements myself, but I'm finding the implementation a little difficult to follow - it'd probably be better off left to someone else.

In any event, we've got a new project in the works (_FinEvol_) and have managed to achieve the same order of magnitude PPO performance as rl_games, while doing some preliminary testing. We'll make the project public in a few days and I'll link to the PPO implementation there if you're interested.

![PPO_MLP_Isaac_Gym_Ant_by_time](https://user-images.githubusercontent.com/72412509/162293240-72753582-8a04-4447-8d0f-807d990234ad.png)

![PPO_MLP_Isaac_Gym_Ant_by_samples](https://user-images.githubusercontent.com/72412509/162293317-d8ba45a3-90bd-4070-9ffe-e4f010b57ff2.png)

Sorry for all the inconvenience,
Momin
